stock_picking_package_preparation_line/models/stock_picking_package_preparation.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2015 Francesco Apruzzese - Apulia Software srl
+# Copyright 2015-2018 Lorenzo Battistini - Agile Business Group
+# Copyright 2016 Alessio Gerace - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api, _
+from datetime import datetime
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class StockPickingPackagePreparation(models.Model):
+
+    _inherit = 'stock.picking.package.preparation'
+
+    FIELDS_STATES = {'done': [('readonly', True)],
+                     'in_pack': [('readonly', True)],
+                     'cancel': [('readonly', True)]}
+
+    picking_type_id = fields.Many2one(related='picking_ids.picking_type_id')
```

Comment:
@SimoRubi readonly = True

Reply:
Done in https://github.com/OCA/stock-logistics-workflow/pull/447/commits/8ed7204eeedc97f7c2cb123f655664a1443e728e